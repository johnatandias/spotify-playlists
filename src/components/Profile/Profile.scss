@import '../../variables';

.profile {
  display: flex;
  flex-direction: column;
  align-items: center;

  span {
    color: $primary-text-color;
  }

  &__avatar {
    width: 130px;
    height: 130px;
    border-radius: 50%;
  }

  &__informations {
    display: flex;
    flex-direction: column;

    &__name {
      font-weight: 500;
      font-size: 20px;
      margin-top: 20px;
    }

    &__account {
      margin-top: 5px;
      font-size: 11px;
    }
  }

  &--minimal {
    flex-direction: row;

    &__avatar {
      height: 60px;
      width: 60px;
    }

    &__informations {
      margin-left: 10px;

      &__name {
        margin-top: 0;
      }
    }
  }
}
