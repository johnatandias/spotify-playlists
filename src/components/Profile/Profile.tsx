import React from 'react';
import './Profile.scss';

interface Profile {
  avatar: string,
  name: string,
  username: string,
  followers: number,
  className?: string,
  minimal?: boolean
}

export const Profile = ({ avatar, name, username, followers, className = '', minimal }: Profile) => (
  <div className={`profile ${minimal ? 'profile--minimal' : ''} ${className}`}>
    <img src={avatar} alt={name} className={`profile__avatar ${minimal ? 'profile--minimal__avatar' : ''}`} />

    <div className={`profile__informations ${minimal ? 'profile--minimal__informations' : ''}`}>
      <span className={`profile__informations__name ${minimal ? 'profile--minimal__informations__name' : ''}`}>
        {name}
      </span>

      <span className={`profile__informations__account ${minimal ? 'profile--minimal__informations__account' : ''}`}>
        {username} | followers: {followers}
      </span>
    </div>
  </div>
);
