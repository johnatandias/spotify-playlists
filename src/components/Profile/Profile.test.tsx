import React from "react";
import { shallow } from 'enzyme';
import { Profile } from './Profile';

describe('Profile component', () => {
  const profile = { avatar: 'image', name: 'Johnatan Dias', username: 'johnatandias', followers: 10 };

  it('Renders without crashing', () => {
    const wrapper = shallow(<Profile {...profile} />);
    expect(wrapper.find('img').prop('src')).toEqual(profile.avatar);
    expect(wrapper.find('img').prop('alt')).toEqual(profile.name);
    expect(wrapper.find('.profile__informations__name').text().includes(profile.name)).toBe(true);
    expect(wrapper.find('.profile__informations__account').text().includes(profile.username)).toBe(true);
    expect(wrapper).toMatchSnapshot();
  });

  it('Render in minimal format', () => {
    const wrapper = shallow(<Profile {...profile} minimal />);
    expect(wrapper.hasClass('profile--minimal')).toBe(true);
    expect(wrapper.find('.profile--minimal__avatar')).toHaveLength(1);
    expect(wrapper.find('.profile--minimal__informations')).toHaveLength(1);
    expect(wrapper.find('.profile--minimal__informations__name')).toHaveLength(1);
    expect(wrapper.find('.profile--minimal__informations__account')).toHaveLength(1);
    expect(wrapper).toMatchSnapshot();
  });
});
