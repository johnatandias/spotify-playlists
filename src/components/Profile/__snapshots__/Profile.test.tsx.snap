// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile component Render in minimal format 1`] = `ShallowWrapper {}`;

exports[`Profile component Renders without crashing 1`] = `ShallowWrapper {}`;
